### PR TITLE
Introduce stable URLs for games

### PR DESCRIPTION
**Pull request by ClementPasteau**
This PR does quite a few things (sorry for the mono-commit!)

In summary:
- improved some components readability + optimisation of network calls
- prevent navigating away from the build dialog at critical moments, resulting in confusing the user when the page is reset (while uploading / while building for web)
- improved the builds list accessible from the Export dialog and from the Games dashboard
  - a web build can be promoted as the latest version for the game
  - a web build can also be unpublished
  - the web build url can be opened directly from the list
  - the list of web builds does not expire any more (only the desktop & mobile builds expire after 7 days)
- change the link of the web build to be:
  - either the stable game page link if the build corresponds to the one linked to the game (especially useful for the first build someone creates, automatically linked to the game)
  - or the build link for subsequent builds, with an option to publish easily to the stable game page
- Added a button to open the stable game page from the game dashboard

Few screenshots:

image
image
image
image
image

### CODE REVIEW

File: newIDE/app/src/GameDashboard/GameRegistration.js [hunk 1]
```diff
@@ -152,9 +152,11 @@ export const GameRegistration = ({
 
   React.useEffect(
     () => {
-      loadGame();
+      if (!game) {
+        loadGame();
+      }
     },
-    [loadGame]
+    [loadGame, game]
```
Comment (4ian) [inline newIDE/app/src/GameDashboard/GameRegistration.js:159]: @ClementPasteau Interesting, why was this necessary?

File: newIDE/app/src/Export/Builds/BuildsDialog.js [hunk 2]
```diff
@@ -1,56 +1,85 @@
...
-          authenticatedUser={authenticatedUser}
-          gameId={gameId}
-        />
-      </Dialog>
-    );
-  }
-}
+
+const BuildsDialog = ({ authenticatedUser, gameId, open, onClose }: Props) => {
+  // Force the Dialog repositioning
```
Comment (4ian) [inline newIDE/app/src/Export/Builds/BuildsDialog.js:23]: Move this comment at the place where forceUpdate is used :)

**Review by 4ian — COMMENTED** [review]
lgtm overall!

File: newIDE/app/src/GameDashboard/GameRegistration.js (hunk 1 above)
Comment (ClementPasteau) [reply newIDE/app/src/GameDashboard/GameRegistration.js:159]: I noticed that the game was fetched multiple times.
I assume that `loadGame` is a callback listening to `profile` and during this process, the limits are checked, so the profile is refetched, so the function change, and so the game is loaded again.
It's not mandatory but it saves unnecessary calls

Comment (4ian) [reply newIDE/app/src/GameDashboard/GameRegistration.js:159]: Perfect then :)

File: newIDE/app/src/Export/Builds/index.js [hunk 3]
```diff
@@ -3,12 +3,15 @@ import React, { Component } from 'react';
...
+import { type Game } from '../../Utils/GDevelopServices/Game';
 import BuildsWatcher from './BuildsWatcher';
 
 type Props = {|
   onBuildsUpdated?: () => void,
   authenticatedUser: AuthenticatedUser,
-  gameId: string,
+  game?: ?Game,
+  onGameUpdated?: Game => void,
+  isGameLoading: boolean,
```
Comment (4ian) [inline newIDE/app/src/Export/Builds/index.js:14]: Nitpicking: can't you consider game === null as loading?

Comment (ClementPasteau) [reply newIDE/app/src/Export/Builds/index.js:14]: No because you could open the builds dialog before you game is registered, and then game will be null.
Or we can decide to hide the button to see the builds, if the game is not registered & loaded

**Review by 4ian — APPROVED** [review]
